airflow/models.py

```diff
@@ -1117,6 +1149,9 @@ def __init__(
         self.pool = pool
         self.sla = sla
         self.execution_timeout = execution_timeout
+        self.on_failure_callback = on_failure_callback
+        self.on_success_callback = on_success_callback
+        self.on_retry_callback = on_retry_callback
```

Comment:
If we are going that route, maybe a dictionary would make more sense ? 
```
{"on_success": callback_callable, "on_failure": notification_callback, "on_retry": email_callback}
```
But I think it's makes the interface less clear.